addons/mass_mailing/models/mailing.py

```diff
@@ -463,6 +465,55 @@ def _group_expand_states(self, states, domain, order):
     # ACTIONS
     # ------------------------------------------------------
 
+    def action_add_favorite(self):
+        """Add the current mailing in the favorites list."""
+        self.ensure_one()
+
+        if self.is_body_empty:
+            return {
+                'type': 'ir.actions.client',
+                'tag': 'display_notification',
+                'params': {
+                    'type': 'warning',
+                    'message': _('Please create a template before adding in your favorites list.'),
+                    'sticky': False,
+                    'next': {'type': 'ir.actions.act_window_close'},
+                }
+            }
+
+        self.write({
```

Comment:
Why not doing 'self.favorite = True' and do a computed field for favorite_date ?